### PR TITLE
Am gestures

### PR DESCRIPTION
**Pull request by amol-c**
Problem Definition:

1. https://trello.com/c/jxbBdnOn/843-double-tap-to-zoom

2. https://trello.com/c/PPWNjiC9/841-gesture-swipe-down-to-close-preview

Solution:

1. Added gesture recognizers on the ScrollView and the PhotoViewer.

### CODE REVIEW

File: Pod/Classes/Extensions/UIScrollView.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    x = zoomPoint.x - zoomSize.width / 2.0
+    y = zoomPoint.y - zoomSize.height / 2.0
+    width = zoomSize.width
+    height = zoomSize.height
+    let zoomRect = CGRect(x: x, y: y, width: width, height: height)
+
+    //apply the resize
+    self.zoomToRect(zoomRect, animated: animated)
+  }
+}
```
Comment (jamesdo) [inline Pod/Classes/Extensions/UIScrollView.swift]: newline

File: Pod/Classes/Protocols/PerspectivePhotoViewer.swift [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+protocol PerspectivePhotoViewer {
+  var photoArray: [PerspectivePhoto]! { get set }
+  var startIndex: Int { get set }
+}
```
Comment (jamesdo) [inline Pod/Classes/Protocols/PerspectivePhotoViewer.swift]: newline

File: Pod/Classes/Views/PerspectivePhotoHolderCell.swift [hunk 3]
```diff
@@ -8,8 +11,25 @@ class PerspectivePhotoHolderCell: UICollectionViewCell {
...
+  }
+
+  private func setupDoubleTapGesture() {
+    let doubleTap = UITapGestureRecognizer(target: self, action: "userDidDoubleTapped:")
+    doubleTap.numberOfTapsRequired = 2
+    self.scrollView.addGestureRecognizer(doubleTap)
+  }
+
+  // MARK: Actions
+  func userDidDoubleTapped(gesture: UITapGestureRecognizer) {
```
Comment (jamesdo) [inline Pod/Classes/Views/PerspectivePhotoHolderCell.swift]: userDidDoubleTap

File: Pod/Classes/Views/PerspectivePhotoHolderCell.swift [hunk 4]
```diff
@@ -8,8 +11,25 @@ class PerspectivePhotoHolderCell: UICollectionViewCell {
...
     self.scrollView.delegate = self
-    self.scrollView.minimumZoomScale = 1.0
+    self.scrollView.minimumZoomScale = MinimumZoomScale
     self.scrollView.maximumZoomScale = 2.0
+
+    self.setupDoubleTapGesture()
+  }
+
+  private func setupDoubleTapGesture() {
+    let doubleTap = UITapGestureRecognizer(target: self, action: "userDidDoubleTapped:")
```
Comment (jamesdo) [inline Pod/Classes/Views/PerspectivePhotoHolderCell.swift]: userDidDoubleTap

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift [hunk 5]
```diff
@@ -48,4 +49,15 @@ public class PerspectivePhotoBrowserViewController: UIViewController, Perspectiv
...
+  // MARK: Actions
+  func swippedVertically() {
+    self.dismissViewControllerAnimated(true, completion: .None)
+  }
+  
+  // MARK: Actions
+  @IBAction func userDidPress(doneButton sender: UIButton) {
+    self.dismissViewControllerAnimated(true, completion: .None)
+  }
+
```
Comment (jamesdo) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: remove extra line

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift [hunk 6]
```diff
@@ -16,9 +11,15 @@ public class PerspectivePhotoBrowserViewController: UIViewController, Perspectiv
...
   public var thumbnailViewController: PerspectiveThumbnailViewController!
 
-  // MARK: Actions
-  @IBAction func userDidPress(doneButton sender: UIButton) {
-    self.dismissViewControllerAnimated(true, completion: .None)
+  // MARK: Overrides
+  public override func viewDidLoad() {
+    super.viewDidLoad()
+
+    let swipeGestureRecognizer = UISwipeGestureRecognizer(target: self, action: "swippedVertically")
```
Comment (jamesdo) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: `:` in the selector? Swiped is also spelled wrong in the selector.

File: Pod/Classes/PerspectivePhotoBrowserViewController.swift [hunk 7]
```diff
@@ -48,4 +49,15 @@ public class PerspectivePhotoBrowserViewController: UIViewController, Perspectiv
       return
     }
   }
+
+  // MARK: Actions
+  func swippedVertically() {
```
Comment (jamesdo) [inline Pod/Classes/PerspectivePhotoBrowserViewController.swift]: swipedVertically()

File: Pod/Classes/Extensions/UIScrollView.swift (hunk 1 above)
Comment (amol-c) [inline Pod/Classes/Extensions/UIScrollView.swift]: Done

**Pull request merged**